tests/unit/file_browser_test.py

```diff
@@ -197,31 +197,26 @@ def directory_filter(path: Path) -> bool:
         return path.is_dir() and not path.name.startswith("_")
 
     BASE_PATH = HERE.parent.parent
-    directory = solara.reactive(BASE_PATH)
 
     @solara.component
     def Page():
-        def protect():
-            def check_base_path(value):
-                if not str(value).startswith(str(BASE_PATH)):
-                    directory.value = BASE_PATH
+        def set_directory(path: Path) -> None:
+            directory.value = path if str(path).startswith(str(BASE_PATH)) else BASE_PATH
```

Comment:
```suggestion
            directory.value = path if str(path).startswith(str(BASE_PATH)) else BASE_PATH
```
Just to make you aware that this will always assign and will always need to do a comparison. Not that it matters because it is a test, but this might be slower in cases where compare is slow

Reply:
Hmm, I never thought about that. But it would also hard to imagine a path comparison could be too expensive. Do you have an example in mind?

Reply:
No, just wanted to make you aware that the behaviour has changed. Happy to merge this as is!